pkgs/development/tools/oq/default.nix

```diff
@@ -0,0 +1,25 @@
+{ lib, fetchFromGitHub, crystal, jq, libxml2 }:
+
+crystal.buildCrystalPackage rec {
+  pname = "oq";
+  version = "0.2.1";
+
+  src = fetchFromGitHub {
+    owner = "Blacksmoke16";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0wc1wdvqybgybq09dr78b93mdyvr7avya5w8kffbv8p4i9yyrfn4";
+  };
+
+  buildInputs = [ jq libxml2 ];
+
+  crystalBinaries.oq.src = "src/oq_cli.cr";
+
```

Comment:
Use `wrapProgram` in postInstall to create a wrapper that includes the $PATH to jq:

```
postInstall = ''
  wrapProgram "$out/bin/oq" \
    --prefix PATH : "${lib.makeBinPath [ jq ]}"
'';
```